packages/rum-core/src/domain/rumSessionManager.ts

```diff
@@ -1,14 +1,16 @@
 import { Configuration, performDraw, startSessionManagement, RelativeTime } from '@datadog/browser-core'
-import { SessionContext } from '../../../core/src/domain/session/sessionManagement'
 import { LifeCycle, LifeCycleEventType } from './lifeCycle'
 
 export const RUM_SESSION_KEY = 'rum'
 
 export interface RumSessionManager {
-  getId: (startTime?: RelativeTime) => string | undefined
-  isTracked: (startTime?: RelativeTime) => boolean
-  hasReplayPlan: (startTime?: RelativeTime) => boolean
-  hasLitePlan: (startTime?: RelativeTime) => boolean
+  findSession: (startTime?: RelativeTime) => RumSession | undefined
+}
+
+export type RumSession = {
+  id: string
+  hasReplayPlan: boolean
+  hasLitePlan: boolean
 }
```

Comment:
I think it would have been better with your proposal in the RFC:
* RumSession has a `isTracked` field
* The `findSession` functions always return an object

Here, `findSession` only returns tracked sessions, which could be confusing. I think we should a single session concept accross the lib, which could be either tracked or not.

Reply:
From my understanding, across the lib there are 3 different session states:

1. tracked, the session has an id and is updated along the user navigation
2. not tracked, the session does not have an id but it is updated along the user navigation
3. inactive, no session in cookie and waiting for sdk to start or an user interaction after expiration

In the current implementation, `isTracked()` or `getId() !== undefined` is used to identify state 1 VS state 2 and 3, since outside session management we don't care about the difference between not tracked and inactive.

It is why I choose to expose only tracked session outside of session management, it allow us to:

- not expose track / no tracked / inactive concepts outside of session management (as long as we don't need to)
- guarantee that we always have an id (for RUM)

With your proposal:

- the findSession functions always return an object
- have a single session concept accross the lib

I think we should then:

- maintain a state for inactive session, in order to always return something for find session
- expose the difference between not tracked and inactive, in order to have a single concept across the lib

IMO 

- `(rum|logs)Manager.findSession` returning tracked session only is simpler for callers
- introducing extra logic / state is not required for current needs

wdyt?

Reply:
FMU, when starting the session manager, we get or create an active session (by [calling `sessionStore.expandOrRenewSession()`](https://github.com/DataDog/browser-sdk/blob/b8d1a67ee92da9d30b3b75b96fd8b157d9c58d2e/packages/core/src/domain/session/sessionManager.ts#L45)). So there can't be any "inactive" session coming from the session manager. This initial session state is stored in the `sessionContextHistory` starting from the `origin` time, so calling `findSession` will always return a session in practice, right?

Reply:
seen IRL, `sessionContextHistory.findSession` can return undefined when the session is expired.

added some clarification on c4e7625